edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs

```diff
@@ -53,9 +57,6 @@ public bool RemovePodInfo(string podName)
                 sinceSeconds: since.Map(sec => (int?)sec).OrDefault(),
                 cancellationToken: cancellationToken);
 
-        public Task<SystemInfo> GetSystemInfo()
-        {
-            return this.moduleManager.GetSystemInfoAsync();
-        }
+        public Task<SystemInfo> GetSystemInfo() => this.moduleManager.GetSystemInfoAsync(this.tokenSource.Token);
```

Comment:
Usually, the caller passes the cancellation token into the method. What will happen if the token will be canceled due to a timeout and caller execute this method?

Reply:
all other methods from the IRuntimeInfoProvider interface have cancellation token as a parameter. we should add it here as well and move cancellation token to the higher level

Reply:
I'm always leery of changing interfaces that haven't changed in a while, but I agree.  It's going to a lot more code changes, though/